### PR TITLE
Support dumping Sygus commands.

### PR DESCRIPTION
**Pull request by abdoo8080**
This initial commit enables the dumping infrastructure to print simple Sygus problems. Most Sygus commands are supported. However, SynthFunCommand needs some improvement.

### CODE REVIEW

File: src/parser/parser.cpp [hunk 1]
```diff
@@ -368,10 +368,10 @@ bool Parser::isUnresolvedType(const std::string& name) {
 }
 
 std::vector<DatatypeType> Parser::mkMutualDatatypeTypes(
-    std::vector<Datatype>& datatypes, bool doOverload) {
+    std::vector<Datatype>& datatypes, bool doOverload, uint32_t flags) {
```
Comment (ajreynol) [inline src/parser/parser.cpp]: This isn't clang-formatted (curly brackets go on their own lines).

Comment (abdoo8080) [reply src/parser/parser.cpp]: Done!

File: src/parser/smt2/Smt2.g [hunk 2]
```diff
@@ -784,7 +784,7 @@ sygusGrammarV1[CVC4::Type & ret,
                             << std::endl;
     }
     std::vector<DatatypeType> datatypeTypes =
-        PARSER_STATE->mkMutualDatatypeTypes(datatypes);
+        PARSER_STATE->mkMutualDatatypeTypes(datatypes, false, ExprManager::DATATYPE_FLAG_PLACEHOLDER);
```
Comment (ajreynol) [inline src/parser/smt2/Smt2.g]: This too doesn't look formatted?

File: src/parser/parser.h [hunk 3]
```diff
@@ -602,9 +602,15 @@ class CVC4_PUBLIC Parser {
    * For each symbol defined by the datatype, if a symbol with name already exists,
    *  then if doOverload is true, we create overloaded operators.
    *  else if doOverload is false, the existing expression is shadowed by the new expression.
+   *
+   * flags specify information about the datatype, e.g. whether it should be printed
+   * out as a definition in models or not
+   *   (see enum in expr_manager_template.h).
```
Comment (abdoo8080) [inline src/parser/parser.h]: Ignored clang-format and followed file convention for this comment.

File: src/parser/smt2/Smt2.g (hunk 2 above)
Comment (abdoo8080) [reply src/parser/smt2/Smt2.g]: Addressed.

**Comment by ajreynol** [comment]
As an internal note, this is work towards https://github.com/CVC4/cvc4-projects/issues/38.

**Pull request merged**